eureka/S3_data_reduction/niriss.py

```diff
@@ -460,53 +187,67 @@ def wave_NIRISS(wavefile, meta):
        The name of the .FITS file with the wavelength
        solution.
     meta : object
+    filename : str, optional
+       The flux filename. Default is None. Needs a filename if
+       the `meta` class is not provided.
 
     Returns
     -------
     meta : object
     """
-    hdu = fits.open(wavefile)
-
-    meta.wavelength_order1 = hdu[1].data + 0.0
-    meta.wavelength_order2 = hdu[2].data + 0.0
-    meta.wavelength_order3 = hdu[3].data + 0.0
-
-    hdu.close()
+    if meta is not None:
+        rampfitting_results = datamodels.open(meta.filename)
+    else:
+        rampfitting_results = datamodels.open(filename)
 
-    return meta
+    # Run assignwcs step on Stage 1 outputs:
+    assign_wcs_results = calwebb_spec2.assign_wcs_step.AssignWcsStep.call(rampfitting_results)
```

Comment:
This step should have already happened in S2, so maybe we should also run `wave_NIRISS()` during S2 to calculate the wavelength solution?  Also, how does this differ from `hdulist['WAVELENGTH',1].data`?

Reply:
Unless I'm totally missing a step (which is possible!) this isn't an output from stage 2 as of right now

Reply:
Strange, this should definitely be happening in Stage 2 - it's a stage 2 function after all

Reply:
I found where this is happening now in Stage 2! Wavelengths are not saved in the calints file.